### PR TITLE
HDDS-11270. [hsync] Add DN layout version (HBASE_SUPPORT/version 8) upgrade test.

### PR DESCRIPTION
**Pull request by jojochuang**
## What changes were proposed in this pull request?
HDDS-11270. [hsync] Add DN layout version (HBASE_SUPPORT/version 8) upgrade test.

Please describe your PR in detail:
This is a test-only PR built on top of HDDS-11259.

Add a test class TestDatanodeUpgradeToHBaseSupport which has two test cases
* PutBlock request with incremental chunk list
* Block finalization request 

These two requests are rejected before upgrade finalizes and accepted after upgrade.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-11270

## How was this patch tested?

Test only code

### CODE REVIEW

**Review by errose28 — COMMENTED** [review]
Thanks for adding this @jojochuang. The new test class looks good overall. Once #7012 is merged and the diff is reduced I can do a full review.

File: hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/upgrade/TestDatanodeUpgradeToHBaseSupport.java
Comment (jojochuang) [reply hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/upgrade/TestDatanodeUpgradeToHBaseSupport.java:122]: Good idea. Added that.

**Comment by jojochuang** [comment]
will rebase this PR against master after the HDDS-7593 feature branch gets merged.

**Review by errose28 — APPROVED** [review]
Thanks for adding this @jojochuang. Just one minor comment otherwise LGTM. If you want to wait until HBase branch is merged into master that's ok.

**Comment by errose28** [comment]
BTW I filed HDDS-11282 to extract the common setup code for these types of upgrade tests.

File: hadoop-hdds/common/src/test/java/org/apache/hadoop/ozone/container/ContainerTestHelper.java
Comment (jojochuang) [reply hadoop-hdds/common/src/test/java/org/apache/hadoop/ozone/container/ContainerTestHelper.java:69]: They don't actually depend on each other, but it's a good suggestion. I'm moving the declaration to OzoneConstants.

**Review by errose28 — APPROVED** [review]

**Comment by jojochuang** [comment]
Merged. Thanks @errose28 !

**Pull request merged**
